### PR TITLE
fix: max_active_session does not work for http handler.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13931)

### CODE REVIEW

File: src/query/service/src/servers/http/v1/query/http_query.rs [hunk 1]
```diff
@@ -235,7 +235,8 @@ impl HttpQuery {
             }
             session
         } else {
-            ctx.get_session(SessionType::HTTPQuery)
+            ctx.upgrade_session(SessionType::HTTPQuery)
```
Comment (BohuTANG) [inline src/query/service/src/servers/http/v1/query/http_query.rs:238]: Better to add a http api unit test to check the max sessions.

**Review by zhang2014 — APPROVED** [review]

**Review by ariesdevil — APPROVED** [review]

File: src/query/service/src/servers/http/v1/query/http_query.rs (hunk 1 above)
Comment (youngsofun) [reply src/query/service/src/servers/http/v1/query/http_query.rs:238]: working on it

Comment (youngsofun) [reply src/query/service/src/servers/http/v1/query/http_query.rs:238]: done

**Comment by youngsofun** [comment]
1. new impl not longer add dummy session to active sessions unless it is later upgraded. otherwise multi concurrent dummy sessions will occupy the slots for normal sessions, especially make it hard to test .
2. add tests

**Pull request merged**